From 44b18e4748b1d5c422d4517648c9bb1267c07bfc Mon Sep 17 00:00:00 2001
From: "Yuriy M. Kaminskiy" <yumkam@gmail.com>
Date: Sat, 9 Feb 2019 18:56:09 +0300
Subject: [PATCH 4/6] Merge back unneeded libgf_util.la

As SIMD_FLAGS contains only preprocessor macros, splitting off
gf_method.c compilation is no longer needed for runtime cpu detection.
---
 src/Makefile.am | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 4949497..5e7dafc 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -5,20 +5,12 @@ AUTOMAKE_OPTIONS = subdir-objects
 
 AM_CPPFLAGS = -I$(top_builddir)/include -I$(top_srcdir)/include
 
-# avoid using SIMD_FLAGS for code that calls strcmp as new gcc
-# versions will use SIMD for the strcmp implementation. Instead
-# we create a static library just for gf_method that is not compiled
-# with SIMD_FLAGS, this static library will get linked into gf_complete.so
-noinst_LTLIBRARIES = libgf_util.la
-libgf_util_la_SOURCES = gf_method.c
-libgf_util_la_CFLAGS = -O3 -fPIC -Wsign-compare
-
-# we narrowly use SIMD_FLAGS for code that needs it
+noinst_LTLIBRARIES =
 lib_LTLIBRARIES = libgf_complete.la
 libgf_complete_la_SOURCES = gf.c gf_wgen.c gf_w4.c gf_w8.c gf_w16.c gf_w32.c \
-          gf_w64.c gf_w128.c gf_rand.c gf_general.c gf_cpu.c
+          gf_w64.c gf_w128.c gf_rand.c gf_general.c gf_cpu.c gf_method.c
 libgf_complete_la_CFLAGS = -O3 $(SIMD_FLAGS) -fPIC -Wsign-compare
-libgf_complete_la_LIBADD = libgf_util.la
+libgf_complete_la_LIBADD =
 
 if HAVE_NEON
 noinst_LTLIBRARIES += libgf_neon.la
-- 
2.11.0

